File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {

        let left = 0, right = numbers.length - 1;

        while(left < right)
        {
            if(numbers[left] + numbers[right] == target) return [left+1, right+1];
            else if(numbers[left] + numbers[right] < target) left += 1;
            else right -= 1;
        }

        return [-1, -1];
    }
}

File: Data Structures & Algorithms/pow-x-n/submission-0.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if(n < 0) return myPow((double)(1/x), n * -1);
        if(n == 0) return 1;
        
        if(n % 2) return x * myPow(x, n-1);
        else return myPow(x, n/2) * myPow(x, n/2);
    }
};
